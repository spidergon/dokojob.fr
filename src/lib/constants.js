export const locations = ['Guadeloupe', 'Guyane', 'Martinique', 'Mayotte', 'Réunion'];

export const contracts = [
  'CDI',
  'CDD',
  'Intérim',
  'Télétravail',
  'Alternance',
  'Saisonnier',
  'Stage',
];

export const codeToLabel = {
  CDI: 'Contrat à durée indéterminée',
  CDD: 'Contrat à durée déterminée',
  Intérim: 'Mission intérimaire',
  Télétravail: 'Télétravail',
  Alternance: 'Alternance',
  Saisonnier: 'Contrat travail saisonnier',
  Stage: 'Stage',
};

export const PRICE1 = 20;
export const PRICE2 = 45;
export const PRICE3 = 75;
export const PRICE4 = 90;

// export const urlPattern =
//   /https\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&\/=]*)/;

export const emailPattern =
  /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
