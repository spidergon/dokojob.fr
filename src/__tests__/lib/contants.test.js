import * as contants from '@lib/constants';

it('renders lib/constants unchanged', () => {
  expect(contants).toMatchSnapshot();
});
