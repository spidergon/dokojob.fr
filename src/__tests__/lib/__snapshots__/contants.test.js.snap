// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders lib/constants unchanged 1`] = `
{
  "PRICE1": 20,
  "PRICE2": 45,
  "PRICE3": 75,
  "PRICE4": 90,
  "codeToLabel": {
    "Alternance": "Alternance",
    "CDD": "Contrat à durée déterminée",
    "CDI": "Contrat à durée indéterminée",
    "Intérim": "Mission intérimaire",
    "Saisonnier": "Contrat travail saisonnier",
    "Stage": "Stage",
    "Télétravail": "Télétravail",
  },
  "contracts": [
    "CDI",
    "CDD",
    "Intérim",
    "Télétravail",
    "Alternance",
    "Saisonnier",
    "Stage",
  ],
  "emailPattern": /\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/,
  "locations": [
    "Guadeloupe",
    "Guyane",
    "Martinique",
    "Mayotte",
    "Réunion",
  ],
}
`;
